Tests/BuildTests/BuildSystemDelegateTests.swift

```diff
@@ -25,7 +25,7 @@ final class BuildSystemDelegateTests: XCTestCase {
             try XCTSkipIf(true, "test is only supported on macOS")
             #endif
             let (fullLog, _) = try executeSwiftBuild(fixturePath)
-            XCTAssertTrue(fullLog.contains("ld: warning: ignoring duplicate libraries: '-lz'"), "log didn't contain expected linker diagnostics")
+            XCTAssertFalse(fullLog.contains("ld: warning: ignoring duplicate libraries: '-lz'"), "log contains unnecessary linker warnings")
```

Comment:
I don't think it's correct to change this test, it was supposed to ensure that linker warnings are properly propagated. We'll need to find a different linker warning that we can check for here or remove the test if appropriate.

Reply:
Thank you @neonichu  I am thinking about removing the test case.

Reply:
I agree, with the current change we're losing test coverage here. This test needs to change to check for a different warning.

Reply:
Talked with @MaxDesiatov and made changes appropriately for the above test case.